src/cmd_line/commands/file.ts

```diff
@@ -84,26 +84,27 @@ export class FileCommand extends node.CommandBase {
       }
     } else {
       // Using a filename, open or create the file
-      this._arguments.name = <string>untildify(this._arguments.name);
+      this._arguments.name =
+        this._arguments.name.indexOf('file://') === 0
+          ? this._arguments.name.slice(7)
+          : <string>untildify(this._arguments.name);
       filePath = path.isAbsolute(this._arguments.name)
         ? this._arguments.name
         : path.join(path.dirname(editorFilePath), this._arguments.name);
 
       if (filePath !== editorFilePath && !(await util.promisify(fs.exists)(filePath))) {
-        // if file does not exist and does not have an extension
-        // try to find it with the same extension
-        if (path.extname(filePath) === '') {
+        if (this._arguments.createFileIfNotExists) {
```

Comment:
I don't yet see the root cause of the bug. Can you walk through the logic for the fix here?

Reply:
In ts/js, it's typical not to provide the file extension of the file on imports. For instance,

```
import { configuration } from '../../src/configuration/configuration';
```

on a `gf`, filePath should equal something like ` ../../src/configuration/configuration`. The expected behaviour is that we should:
1) try to find the file as is (configuration)
2) if it doesn't have an extension, try to find it with the same extension as the current file (configuration.ts)
3) then if `createFileIfNotExists=true`, we create the file.

WIth your proposed changed, we would always be creating a new file (presuming `createFileIfNotExists=true`). IOW, I think we need another check following the if block of 94 to see if we found it with the same extension.

So maybe something like this:

```
      // Using a filename, open or create the file
      this._arguments.name = <string>untildify(this._arguments.name);
      filePath = path.isAbsolute(this._arguments.name)
        ? this._arguments.name
        : path.join(path.dirname(editorFilePath), this._arguments.name);

      isFound = false;
      if (filePath !== editorFilePath) {
        isFound |= await util.promisify(fs.exists)(filePath)

        if (!isFound) {
            // if file does not exist and does not have an extension
            // try to find it with the same extension
            if (path.extname(filePath) === '') {
              const pathWithExt = filePath + path.extname(editorFilePath);
              isFound |= await util.promisify(fs.exists)(pathWithExt)

              if (isFound) {
                  filePath = pathWithExt;
              }
          }
        }

        // if not found, and createFileIfNotExists = true, create file
        // else if not found, error
}
```

Reply:
Thank you! You are right. Actually, my thoughts about a role of the createFileIfNotExists flag were slightly different.  I've just recovered the logic you are described.

PS: my congratulations :) best wishes for you